manifests/record/mx.pp

```diff
@@ -1,15 +1,36 @@
 # == Define: dns::record::mx
 #
-# Wrapper for dns::record to set an XM record.
+# Wrapper for dns::record to set an MX record.
 #
 define dns::record::mx (
   $zone,
   $data,
-  $ttl = '',
-  $preference = '0',
-  $host = $name ) {
+  $ttl        = '',
+  $preference = 10,
+  $host       = '@' ) {
 
-  $alias = "${host},MX,${zone}"
+  $alias = "${host},${zone},MX,${preference},${data}"
+
+  unless is_string($zone) and is_domain_name($zone) and !is_ip_address($zone) and $zone !~ /\.?\d+$/ {
```

Comment:
I agree that is_domain_name() should cover it, except that function from puppetlabs-stdlb currently returns true for IP addresses, and for numeric top-level domains, which are never domain names. This prompted me to add the !is_ip_address condition, the not-a-digit regex, and open [PUP-3856](https://tickets.puppetlabs.com/browse/MODULES-1670). Once is_domain_name is more strict, the last two conditions here could be dropped.